yt_dlp/extractor/nhk.py

```diff
@@ -591,102 +592,198 @@ class NhkRadiruIE(InfoExtractor):
+            for i, act in enumerate(acts):
+                res = f'【{role}】' if i == 0 and role is not None else ''
+                if title := act.get('title'):
+                    res += f'{title}…'
+                res += act.get('name')
+                formatted_roles.append(res)
+        return join_nonempty(*formatted_roles, delim='，')
+
+    def _fetch_artists(self, by_artist):
+        if not by_artist or len(by_artist) == 0:
+            return None
+        artists = []
+        for artist in by_artist:
+            res = ''
+
+            name = artist.get('name')
+            role = artist.get('role')
+            part = artist.get('part')
+
+            if role != '':
+                res += f'{role}…'
+            if part != '':
+                res += f'（{part}）'
+            res += name
+
+            if res != '':
+                artists.append(res)
+        if len(artists) == 0:
+            return None
+        return '、'.join(artists)
+
+    def _fetch_duration(self, duration):
+        d = parse_duration(duration)
+        if d is None:
+            return None
+        hours, remainder = divmod(d, 3600)
+        minutes, seconds = divmod(remainder, 60)
+        res = '（'
+        if hours > 0:
+            res += f'{int(hours)}時間'
+        if minutes > 0:
+            res += f'{int(minutes)}分'
+        res += f'{int(seconds):02}秒）'
+        return res
+
+    def _format_musiclist(self, music_list):
+        if not music_list or len(music_list) == 0:
+            return None
+
+        tracks = []
+
+        for track in music_list:
+            track_details = []
+
+            if name := track.get('name'):
+                track_details.append(f'「{name}」')
+            if lyricist := track.get('lyricist'):
+                track_details.append(f'{lyricist}:作詞')
+            if composer := track.get('composer'):
+                track_details.append(f'{composer}:作曲')
+            if arranger := track.get('arranger'):
+                track_details.append(f'{arranger}:編曲')
+            track_details.append(self._fetch_artists(track.get('byArtist')))
+            track_details.append(self._fetch_duration(track.get('duration')))
+
+            if track.get('label') or track.get('code'):
+                track_details.append('＜' + join_nonempty('label', 'code', delim=' ', from_dict=track) + '＞')
+
+            if location := track.get('location'):
+                track_details.append(f'～{location}～')
+
+            tracks.append(join_nonempty(*track_details, delim='\n'))
+
+        return '\n\n'.join(tracks)
+
+    def _format_description(self, response):
+        act = traverse_obj(response, ('misc', 'actList', {self._format_actlist}))
+        music = traverse_obj(response, ('misc', 'musicList', {self._format_musiclist}))
+        desc = join_nonempty('epg80', 'epg200', delim='\n\n', from_dict=traverse_obj(response, 'detailedDescription'))
+
+        return join_nonempty(desc, act, music, delim='\n\n')
```

Comment:
- adjustments based on above suggestions to the `_format_*` methods
- in `join_nonempty`, if `from_dict` is `None`, then it will join the literal strings `epg80` and `epg200`, so we need to typecheck for `{dict}` and fallback with `or {}`
- can inline in the outer `join_nonempty` call
```suggestion
    def _format_description(self, response):
        detailed_description = traverse_obj(response, ('detailedDescription', {dict})) or {}
        return join_nonempty(
            join_nonempty('epg80', 'epg200', delim='\n\n', from_dict=detailed_description),
            traverse_obj(response, ('misc', 'actList', {self._format_act_list})),
            traverse_obj(response, ('misc', 'musicList', {self._format_music_list})),
            delim='\n\n')
```